pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,409 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
-	"log"
 	"math/rand"
 	"net/http"
+	"net/http/httptest"
 	"net/url"
 	"os"
+	"os/exec"
 	"path/filepath"
 	"reflect"
+	"strconv"
 	"strings"
-	"testing"
-	"time"
 
+	"syscall"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/containerized-data-importer/pkg/common"
 	"kubevirt.io/containerized-data-importer/pkg/image"
+	"kubevirt.io/containerized-data-importer/pkg/util"
 	"kubevirt.io/containerized-data-importer/tests/utils"
 )
 
 const TestImagesDir = "../../tests/images"
 
+var imageDir, _ = filepath.Abs(TestImagesDir)
+var cirrosData, _ = readFile(filepath.Join(imageDir, "cirros-qcow2.img"))
```

Comment:
Is there any concern that the test container may run out of memory due to reading entire images into variables?

Reply:
The image are 13 and 18M and we are manipulating and reading them tons of times during the test run. So if an extra 31M of memory is a concern, we have bigger issues.

Reply:
agree. May become an issue if we test much larger (or many) images...

Reply:
I would argue we shouldn't do that in unit tests, that would be an e2e test.